plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -688,7 +697,14 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                      self->log_trx_results( trx, nullptr, ex, 0, start, is_transient );
                      next( std::move(ex) );
                      self->_idle_trx_time = fc::time_point::now();
-                     self->_time_tracker.add_fail_time(self->_idle_trx_time - start);
+                     auto dur = self->_idle_trx_time - start;
+                     if ( is_transient ) {
+                        // this function can only be on main thread.
```

Comment:
```suggestion
                        assert(is_on_main_thread());
```

Reply:
Thanks. The code is inside `app().executor().post` lambda and is on main thread. I will remove the redundant comment there.

Reply:
even better, add the assert so that it documents what you know is true.

Reply:
In this case, it seems an unnecessary assert. As we post to main, we would expect it is on main. All of our app().post() code assumes this.